src/axom/core/Array.hpp

```diff
@@ -550,121 +830,303 @@ Array< T >::~Array()
   m_data = nullptr;
 }
 
+//------------------------------------------------------------------------------
+template< typename T >
+inline void Array< T >::fill( const T& value )
+{
+  for (IndexType i = 0 ; i < m_num_elements ; i++)
```

Comment:
A few things for ``Array::fill``:

- It should be also templated on the desired execution space. The second template argument should be optional and default to ``axom::SEQ_EXEC``
-  Then, implement this using ``axom::for_all()``
- It would be good to add some sanity check to ensure the specified execution space makes sense for the underlying allocator_id (i.e., memory space) for the Array (we may need some more infrastructure to do this).